src/k8s/api/v1/types.go

```diff
@@ -43,8 +43,8 @@ type NodeStatus struct {
 }
 
 type Datastore struct {
-	Type        string `json:"type,omitempty"`
-	ExternalURL string `json:"external-url,omitempty" yaml:"external-url,omitempty"`
+	Type            string   `json:"type,omitempty"`
+	ExternalServers []string `json:"external-servers,omitempty" yaml:"external-servers,omitempty"`
```

Comment:
> There is IMHO also no point in perfectly matching the internal representation of configuration with the external-facing one. They (might) eventually diverge in the future if we introduce new features and new API versions

+1 to this. The `pkg/k8sd/types.ClusterConfig` can diverge at will and is not user-facing, so we can handle it as we see fit for the code. The only type users will be aware of is `api/v1.$Foo`.

Note that this is still API

Reply:
i guess i was asking about the `json` and `yaml` keys being named `external-*`  i had assumed these types would resolve to something that is visible to the users.  So long as they are all consistently named, it's good with me.

Reply:
Alright, I read of the comments, that we can leave it like this.

Reply:
So, back to this specific line -- since it's in `api/v1/types` -- it's not an internal structure?

* can we have this called "servers" in the json and yaml marshalling?  
* is there any need for yaml annotation on the type

Reply:
true - updated